warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -389,14 +392,31 @@ protected IndexInfo intersect(long maxPossibilities, Iterable<IndexMatch> matchI
         }
         
         JexlNode newNode;
-        if (matches.size() > 1) {
+        if (matches.isEmpty()) {
+            JexlNodeSet nodes = new JexlNodeSet();
+            if (matchNode != null) {
+                nodes.add(matchNode);
+            }
+            if (!otherInfiniteNodes.isEmpty()) {
```

Comment:
The check for empty is unnecessary

Reply:
The method is now obsolete. I'm removing it in the next commit.